src/extensions/http_api/definitions.js

```diff
@@ -125,11 +125,18 @@ exports.install = ({ baseUrl = '' } = {}) => {
         this.httpApiClient.setQuery(Cast.object(this.state.populateObject(step.rowsHash())))
     })
 
-    Given(/^(?:I )?pick response json (.+) as (.+)$/, function (path, key) {
+    /**
+     * Pick a value from previous json response and set it to state
+     */
+    Given(/^(?:I )?pick response json (body|headers) (.+) as (.+)$/, function (
+        dataSource,
+        path,
+        key
+    ) {
         const response = this.httpApiClient.getResponse()
-        const body = response.body
+        let data = dataSource === 'body' ? response.body : response.headers
```

Comment:
We can reverse this condition to use body by default.

Reply:
Yes, that's much better. Done!